### PR TITLE
Expose the proxy bypass option

### PR DESCRIPTION
**Pull request by grahamhar**
In some scenarios there will be a mixture of requests some will need a proxy some won't. This adds and exposes the proxy bypass option.

I have only tested this in chrome on linux, can you help by suggesting:

- What tests need to be added.
- Documentation to be added.
- Updates for other browsers.

### CODE REVIEW

**Comment by mdmintz** [comment]
There are some things that should be fixed first:

You need to ``flake8`` all your code and make sure there are no issues there. I see some long lines that would fail right away.

You left debug statements in your code:
``logging.exception('Big issue from GRAHAM')``

You mixed args by inserting ``proxy_bypass_list`` in the middle of grouped args. It should go after ``proxy_pass``.
```python
    proxy_string,
    proxy_bypass_list,
    proxy_auth,
    proxy_user,
    proxy_pass,
```

**Comment by grahamhar** [comment]
I think I addressed all those points @mdmintz

File: seleniumbase/core/browser_launcher.py [hunk 1]
```diff
@@ -1679,6 +1693,7 @@ def get_local_driver(
                 headless,
                 locale_code,
                 proxy_string,
+                proxy_bypass_list,
                 proxy_auth,
                 proxy_user,
                 proxy_pass,
```
Comment (mdmintz) [inline seleniumbase/core/browser_launcher.py:1699]: @grahamhar Looks like you still have one or two more places where you have to fix the ordering

File: seleniumbase/core/browser_launcher.py [hunk 2]
```diff
@@ -1827,6 +1843,7 @@ def get_local_driver(
                         headless,
                         locale_code,
                         proxy_string,
+                        proxy_bypass_list,
                         proxy_auth,
                         proxy_user,
                         proxy_pass,
```
Comment (mdmintz) [inline seleniumbase/core/browser_launcher.py:1849]: @grahamhar Looks like you still have one or two more places where you have to fix the ordering

File: seleniumbase/core/browser_launcher.py (hunk 1 above)
Comment (grahamhar) [reply seleniumbase/core/browser_launcher.py]: fixed

File: seleniumbase/core/browser_launcher.py (hunk 2 above)
Comment (grahamhar) [reply seleniumbase/core/browser_launcher.py]: fixed

**Comment by mdmintz** [comment]
@grahamhar According to https://www.chromium.org/developers/design-documents/network-settings , it's a semi-colon separated list, not a comma-separated one. Have you tried an example with multiple domains bypassed?

```
--proxy-bypass-list=(<trailing_domain>|<ip-address>)[:<port>][;...]

This tells chrome to bypass any specified proxy for the given semi-colon-separated list of hosts. This flag must be used (or rather, only has an effect) in tandem with --proxy-server.
Note that trailing-domain matching doesn't require "." separators so "*google.com" will match "igoogle.com" for example.

For example,
  --proxy-server="foopy:8080" --proxy-bypass-list="*.google.com;*foo.com;127.0.0.1:8080"
will use the proxy server "foopy" on port 8080 for all hosts except those pointing to *.google.com, those pointing to *foo.com and those pointing to localhost on port 8080. 
igoogle.com requests would still be proxied. ifoo.com requests would not be proxied since *foo, not *.foo was specified. 
```

**Comment by mdmintz** [comment]
@grahamhar It's mostly for the Python-comment description, but I'd check for a semicolon-separated list, as a comma could appear in valid urls such as ``data:,Hello``.

**Comment by grahamhar** [comment]
> @grahamhar It's mostly for the Python-comment description, but I'd check for a semicolon-separated list, as a comma could appear in valid urls such as `data:,Hello`.

I think the bypass list only covers domains and IP address not URL so commas are not actually valid. I'll update the comment to reflect it should be a semi colon separated list.

File: seleniumbase/core/browser_launcher.py [hunk 3]
```diff
@@ -807,6 +815,7 @@ def get_driver(
             locale_code,
             servername,
             proxy_string,
+            proxy_bypass_list,
             proxy_auth,
             proxy_user,
             proxy_pass,
```
Comment (mdmintz) [inline seleniumbase/core/browser_launcher.py:821]: Looks like ``proxy_bypass_list`` is not in the correct location here. Should be after ``proxy_pass``.

File: seleniumbase/core/browser_launcher.py [hunk 4]
```diff
@@ -1731,6 +1746,7 @@ def get_local_driver(
                 headless,
                 locale_code,
                 proxy_string,
+                proxy_bypass_list,
                 proxy_auth,
                 proxy_user,
                 proxy_pass,
```
Comment (mdmintz) [inline seleniumbase/core/browser_launcher.py:1752]: Looks like ``proxy_bypass_list`` is not in the correct location here. Should be after ``proxy_pass``.

**Comment by mdmintz** [comment]
@grahamhar I added more comments. Looks like there are still a few places where ``proxy_bypass_list`` was not in the correct location in ``browser_launcher.py``.

**Comment by grahamhar** [comment]
> @grahamhar I added more comments. Looks like there are still a few places where `proxy_bypass_list` was not in the correct location in `browser_launcher.py`.

Sorry, not sure how I missed them. Updated now.

**Comment by mdmintz** [comment]
@grahamhar On your line 943 of ``browser_launcher.py``, it looks like you're calling ``chrome_options = _set_chrome_options(`` without the ``proxy_bypass_list`` as a variable, which breaks my internal tests that include selenium grid tests. Those tests are private because they include private authentication codes to private proxy servers, but I'm happy to walk through all issues I encounter like this.

**Comment by grahamhar** [comment]
@mdmintz I think there will possibly be quite a few gaps like this. I only managed to test using the pycharm fixture. Let me know any more changes and I'll try to update.

**Comment by mdmintz** [comment]
@grahamhar Sounds good. You'll need to add that into ``get_remote_driver(`` to make that new line also work.

**Comment by mdmintz** [comment]
@grahamhar  Your current line 1017 of ``browser_launcher.py`` is missing the ``proxy_bypass_list`` in ``firefox_options = _set_firefox_options(``.

**Comment by mdmintz** [comment]
@grahamhar Merged! Nice work. It'll be part of the next release.

**Comment by mdmintz** [comment]
@grahamhar It has been released as part of https://github.com/seleniumbase/SeleniumBase/releases/tag/v2.2.8

**Pull request merged**